### PR TITLE
Add description for typescript facet and build script

### PR DESCRIPTION
**Pull request by daogrady**
1. Adds a description for how to use `cds add typescript`
2. Adds a description for using `cds build` with TypeScript (which replaces the steps that formerly had to be done manually)
3. Updates the output for `cds-typer --help`

### CODE REVIEW

File: node.js/typescript.md [hunk 1]
```diff
@@ -95,7 +99,9 @@ Run your Jest tests with preset `ts-jest` without precompiling TypeScript files.
     jest
     ```
 
+## Building TypeScript Projects
 
+A dedicated build task for `cds-build` is provided as part of the `cds-typer` package. See [there](../tools/cds-typer#integrate-into-your-multitarget-application) for more information on how to customize this task.
```
Comment (renejeglinsky) [inline node.js/typescript.md:104]:
```suggestion
A dedicated build task for `cds-build` is provided as part of the `cds-typer` package. See [Integrate Into Your Multitarget Application](../tools/cds-typer#integrate-into-your-multitarget-application) for more information on how to customize this task.
```

File: node.js/typescript.md [hunk 2]
```diff
@@ -113,7 +117,9 @@ Run your Jest tests with preset `ts-jest` without precompiling TypeScript files.
     jest
     ```
 
+## Building TypeScript Projects
 
+A dedicated build task for `cds-build` is provided as part of the `cds-typer` package. See [Integrate Into Your Multitarget Application](../tools/cds-typer#integrate-into-your-multitarget-application) for more information on how to customize this task.
```
Comment (LotharBender) [inline node.js/typescript.md:122]:
```suggestion
A dedicated build task for `cds build` is provided as part of the `cds-typer` package. See [Integrate Into Your Multitarget Application](../tools/cds-typer#integrate-into-your-multitarget-application) for more information on how to customize this task.
```

**Review by LotharBender — CHANGES_REQUESTED** [review]

**Review by LotharBender — APPROVED** [review]

File: tools/cds-typer.md [hunk 3]
```diff
@@ -351,21 +356,21 @@ npx @cap-js/cds-typer "*" --outputDirectory @cds-models
 Make sure to add the quotes around the asterisk so your shell environment does not expand the pattern.
 
 ## Integrate Into Your Multitarget Application
-Similar to the integration in your CI, you need to add `cds-typer` to the build process of your MTA file as well.
+Having `cds-typer` present as dependency provides a build task "`typescript`" which is automatically included as part of `cds build` if your project contains a `tsconfig.json`.
```
[the remark this answers is not in the document]
Comment (daogrady) [reply tools/cds-typer.md:401]: Not sure I understand correctly. If you are referring to the [former build task script](https://cap.cloud.sap/docs/tools/cds-typer#integrate-into-your-multitarget-application), then you are correct that it should still work. But I believe it is now completely redundant, as the automatic build task achieves the same thing.
Not sure what the overarchig CAPire-philosophy on that is, but personally, I try to refrain from promoting TIMTOWTDI whenever possible for the sake of clarity.

File: node.js/typescript.md [hunk 4]
```diff
@@ -21,10 +21,14 @@ Follow these steps to add TypeScript support:
     npm i -g typescript ts-node
     ```
 
-2. Add a _tsconfig.json_ file to your project.
+2. Add a _tsconfig.json_ file to your project. A basic _tsconfig.json_ can be generated by running:
```
Comment (renejeglinsky) [inline node.js/typescript.md:24]:
```suggestion
2. Add a basic _tsconfig.json_ file to your project:
```

File: tools/cds-typer.md [hunk 5]
```diff
@@ -399,21 +398,21 @@ npx @cap-js/cds-typer "*" --outputDirectory @cds-models
 Make sure to add the quotes around the asterisk so your shell environment does not expand the pattern.
 
 ## Integrate Into Your Multitarget Application
-Similar to the integration in your CI, you need to add `cds-typer` to the build process of your MTA file as well.
+Having `cds-typer` present as dependency provides a build task "`typescript`" which is automatically included as part of `cds build` if your project has a dependency on `typescript`.
```
Comment (renejeglinsky) [inline tools/cds-typer.md:401]: I don't understand how this relates to MTA. Where is the connection?

Comment (daogrady) [reply tools/cds-typer.md:401]: Sorry, maybe I misunderstood the MTA build. Doesn't `cds build` prepare all files to ultimately create a `.mtar` file, which is then used for MTA deployment?

Comment (renejeglinsky) [reply tools/cds-typer.md:401]: Well, I'd say you're right, but MTA build is not the only way to use `cds build` so I'd expect either a different (more generic) title or a hint towards MTA deployment and maybe a link to the deploy guide.

File: tools/cds-typer.md [hunk 6]
```diff
@@ -399,21 +398,21 @@ npx @cap-js/cds-typer "*" --outputDirectory @cds-models
 Make sure to add the quotes around the asterisk so your shell environment does not expand the pattern.
 
 ## Integrate Into Your Multitarget Application
```
Comment (daogrady) [inline tools/cds-typer.md:400]: @renejeglinsky I see, that makes sense, sense for pointing it out! What do you think about this section title instead?
```suggestion
## Integrate Into Your Build Process
```

Comment (renejeglinsky) [reply tools/cds-typer.md:400]: Sounds good to me :)

File: node.js/typescript.md [hunk 7]
```diff
@@ -113,7 +117,9 @@ Run your Jest tests with preset `ts-jest` without precompiling TypeScript files.
     jest
     ```
 
+## Building TypeScript Projects
 
+A dedicated build task for `cds build` is provided as part of the `cds-typer` package. See [Integrate Into Your Multitarget Application](../tools/cds-typer#integrate-into-your-multitarget-application) for more information on how to customize this task.
```
Comment (renejeglinsky) [inline node.js/typescript.md:122]:
```suggestion
A dedicated build task for `cds build` is provided as part of the `cds-typer` package. See [Integrate Into Your Multitarget Application](../tools/cds-typer#integrate-into-your-build-process) for more information on how to customize this task.
```

File: tools/cds-typer.md (hunk 5 above)
Comment (renejeglinsky) [reply tools/cds-typer.md:401]:
```suggestion
Having `cds-typer` present as dependency provides a build task "`typescript`". If your project, in addition, has a dependency on `typescript,` this build tasks is automatically included as part of `cds build`.
```

File: tools/cds-typer.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (daogrady) [reply tools/cds-typer.md]: Is this still a blocker? Not entirely opposed to leaving it in if you prefer to. 🙂 If you have any more feedback, do let me know

**Review by renejeglinsky — APPROVED** [review]

File: tools/cds-typer.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (renejeglinsky) [reply tools/cds-typer.md]: No not  a blocker :)

File: tools/cds-typer.md [hunk 8]
```diff
@@ -379,41 +378,41 @@ You can safely remove and recreate the types at any time.
...
-  before-all:
-  - builder: custom
-    commands:
-    - npx cds build --production
-    - npx @cap-js/cds-typer "*" --outputDirectory gen/srv/@cds-models
+```json [tsconfig.cdsbuild.json]
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./gen/srv",
```
Comment (LotharBender) [inline tools/cds-typer.md:409]: `outDir` seems to match the default value and might not need to be explicitly configured: `path.join(cds.env.build.target, cds.env.folders.srv)`

Comment (daogrady) [reply tools/cds-typer.md:409]: 🤔 I don't think we can dynamically add values here, as this is the recommendation for the basic layout of a tsconfig.json, which is a static file.

Comment (LotharBender) [reply tools/cds-typer.md:409]: I thought the `typescript` build task could calculate the default path if `outDir` isn't set. 
Of course, this is only an option if `outDir` isn't used by the typescript compiler itself which seems to be the case.

**Pull request merged**